### PR TITLE
Fix the updates explorer and other minor bugs

### PR DESCRIPTION
**Pull request by code-withAshish**
This pull Request fixes (possibly) the update explorer and also some improvements have been made such as:

- Update explorer now used `Virtual Documents` so no need to access the user filesystem, as it is handled by VSC itself hence making it more safe and robust.
- A `CodeLens` has been provided in the Update explorer for users to stop the bot right in their editor so that they dont go on funding the stop button in the sidebar ( that still exists though).
- Improvement in `Filter Query` fuzzy search, it will put relevant searches higher up in the dropdown so the exact match is always on the top.
- The `Filter Query Hover` now works for every query and shows documentation as expected .
- And some file changes and build config improvements that's not much relevant to the end-user but will help in the maintainence of the project in general.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,15 +1,15 @@
 {
-  "name": "grammyjs",
+  "name": "vscode-grammy",
```
Comment (roj1512) [inline package.json:2]: Love that!

File: src/commands/commands.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  StopExplorer = "updates-explorer.stop",
+}
+
+export default function registerCommands(context: vscode.ExtensionContext) {
+  const startCommand = vscode.commands.registerCommand(
+    Commands.StartExplorer,
+    async () => {
+      const token = await vscode.window.showInputBox({
+        title: "Enter your Bot token",
```
Comment (roj1512) [inline src/commands/commands.ts:18]:
```suggestion
        title: "Enter your bot token",
```

File: src/commands/commands.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      });
+      if (token) {
+        await initUpdatesExplorer(token);
+      } else {
+        vscode.window.showErrorMessage("Token was not provided");
```
Comment (roj1512) [inline src/commands/commands.ts:23]:
```suggestion
        vscode.window.showErrorMessage("No bot token was specified.");
```

File: src/updates_explorer/bot_listener.ts [hunk 4]
```diff
@@ -66,20 +61,17 @@ export class BotListener {
           "chat_member",
           "chat_join_request",
         ],
-        onStart: async () => {
-          this.state = "listening";
-        },
+        onStart: async () => {},
```
Comment (roj1512) [inline src/updates_explorer/bot_listener.ts:64]: Can you do this to all empty blocks?

```suggestion
        onStart: async () => {
          //
        },
```

Comment (code-withAshish) [reply src/updates_explorer/bot_listener.ts:64]: Sure thing

**Comment by code-withAshish** [comment]
closing in favour of new version re-write

**Pull request closed without merging**